tests/TestCIT.py

```diff
@@ -1,26 +1,41 @@
-from itertools import chain, combinations
+# from itertools import chain, combinations
+import unittest
 
 import numpy as np
+from causallearn.utils.cit import CIT
+# from causallearn.utils.cit import (chisq, chisq_notoptimized, gsq, gsq_notoptimized)
 
-from causallearn.utils.cit import (chisq, chisq_notoptimized, gsq,
-                                   gsq_notoptimized)
 
+class TestCIT(unittest.TestCase):
+    def test_fisherz_singularity_problem(self):
+        X1 = X2 = np.random.normal(size=1000)
+        X = np.array([X1, X2])
 
-def test_new_old_gsq_chisq_equivalent(self):
-    def powerset(iterable):
-        return chain.from_iterable(combinations(list(iterable), r) for r in range(len(iterable) + 1))
+        cit = CIT(data=X, method='fisherz')
 
-    def _unique(column):
-        return np.unique(column, return_inverse=True)[1]
+        try:
+            cit.fisherz(0, 1, tuple())
+        except ValueError:
+            print('Catch Singularity Problem')
+            return
 
-    data_path = "data_discrete_10.txt"
-    data = np.loadtxt(data_path, skiprows=1)
-    data = np.apply_along_axis(_unique, 0, data).astype(np.int32)
-    cardinalities = np.max(data, axis=0) + 1
+        assert False
 
-    for X in range(data.shape[1]):
-        for Y in range(X + 1, data.shape[1]):
-            for S in powerset([_ for _ in range(data.shape[1]) if _ != X and _ != Y]):
-                assert np.isclose(gsq(data, X, Y, S, cardinalities), gsq_notoptimized(data, X, Y, S))
-                assert np.isclose(chisq(data, X, Y, S, cardinalities), chisq_notoptimized(data, X, Y, S))
-                print(f'{X};{Y}|{S} passed')
+# def test_new_old_gsq_chisq_equivalent(self):

```

Comment:
what's the reason to remove this test?

Reply:
The functions chisq_notoptimized and gsq_notoptimized have been commented out.

Reply:
Yeah, why do you comment out this test?

In the main branch, the test exists: https://github.com/cmu-phil/causal-learn/blob/main/tests/TestCIT.py

Reply:
The functions `chisq_notoptimized` and `gsq_notoptimized` have been commented out in `causallearn/utils/cit.py`. I am not sure if I want to remove this test from the code. But I need to comment out the code in order to test my code.

Reply:
@zhi-yi-huang Cool, now things is clear. Sorry I didn't read your previous message clearly. For some reason I thought chisq_notoptimized() is in the this test lol

I think @MarkDana should have context. Do you mind taking a look? Did you remove chisq_notoptimized() and forget to remove the test?

Normally, in our code base, we shouldn't keep lots of commented out code --- this will make our code more crowded without providing additional value.

I would suggest if chisq_notoptimized() is removed, we can just remove all the code related to it here.

Reply:
Yes the `chisq_notoptimized` was removed by me in the recent commit of CIT class.
It was intended to ensure the code consistency of chisq test w/ or w/o optimization.
And I would suggest to remove the related test here too. Thanks @zhi-yi-huang and @tofuwen !